3ds Max/Max2Babylon/Exporter/BabylonExporter.Skeleton.cs

```diff
@@ -135,10 +110,88 @@ private List<IIGameNode> GetRevelantNodes(IIGameSkin skin)
                 }
             }
 
-            revelantNodes = sorted;
-            revelantNodesBySkin.Add(skin, revelantNodes);   // Stock the result for optimization
+            revelantNodesBySkin.Add(skin, sorted);   // Stock the result for optimization
+
+            return sorted;
+        }
+
+        private IIGameNode GetCommonAncestor(List<IIGameNode> bones)
+        {
+            IIGameNode ancestor = null;
+            foreach (IIGameNode b in bones)
+            {
+                ancestor = GetCommonAncestor(ancestor, b);
+                if (ancestor == null)
+                {
+                    break;
+                }
+            }
+
+            return ancestor;
+        }
+
+        private IIGameNode GetCommonAncestor(IIGameNode nodeA, IIGameNode nodeB)
+        {
+            if (nodeA == nodeB || nodeB == null) return nodeA;
+            if (nodeA == null) return nodeB;
+
+            List<IIGameNode> previousANodes = new List<IIGameNode>();
+            previousANodes.Add(nodeA);
+            List<IIGameNode> previousBNodes = new List<IIGameNode>();
+            previousBNodes.Add(nodeB);
+
+            int previousANodesIndex = 0;
+            int previousBNodesIndex = 0;
+
+            while (true)
+            {
+                if (previousANodes[previousANodesIndex].NodeParent != null)
+                {
+                    previousANodes.Add(previousANodes[previousANodesIndex].NodeParent);
+                    ++previousANodesIndex;
+                }
+
+                if (previousBNodes[previousBNodesIndex].NodeParent != null)
+                {
+                    previousBNodes.Add(previousBNodes[previousBNodesIndex].NodeParent);
+                    ++previousBNodesIndex;
+                }
+
+                if (previousANodes[previousANodesIndex].NodeParent == null && previousBNodes[previousBNodesIndex].NodeParent == null)
+                {
+                    break;
+                }
+
+
+            }
+
+            IIGameNode topA = previousANodes[previousANodesIndex--];
+            IIGameNode topB = previousBNodes[previousBNodesIndex--];
+            if (topA.MaxNode.Handle != topB.MaxNode.Handle)
+            {
+                return null;
+            }
+
+            
+
+            while (true)
+            {
+                // in the case that one is a child of the other, other will be invalid when one still has more nodes to go
```

Comment:
Here's where the magic happens, and we walk the hierarchy to find the shared ancestor.

if we have a tree like:
![image](https://user-images.githubusercontent.com/3952729/56828916-3ab52f80-6817-11e9-9953-3c99f00a398a.png)

And our inputs are D and F, do we expect to return A or C?
Given the function name "GetCommonAncestor", I'd assume that we want C

Likewise, for a case of F and C, our expected output should be C as well, right?

Can we create a scene to test this functionality?

Reply:
Yes, C. In fact in the original code the name is GetLowestCommonAncestor. It is a problem in our engine for the skeleton hierarchy to contain the entire chain up to A, because of culling. Our `skin` object is attached to the node just above the skeleton root. If B was animated to move a long distance, and the skeleton root is on or above A, the bounding box of the skin would not move with the animation of B. This results in the object disappearing when you do not look at the t-pose position of the bounding box, even though there is not even any skinning or skinned mesh on or above B.